R/process.R

```diff
@@ -430,44 +434,79 @@ process_generate_id <- function(x, prefix, sort = FALSE) {
 #'
 #' @param my_list List of input information
 #' @param dataset_id Identifier for a particular study in the AusTraits database
+#' @param traits Table of trait data (for this function, just the data.csv file with custom_R_code applied)
 #' @return Tibble with context details if available
 #' @importFrom rlang .data
 #'
 #' @examples
 #' \dontrun{
-#' process_format_contexts(read_metadata("data/Apgaua_2017/metadata.yml")$context)
+#' process_format_contexts(read_metadata("data/Apgaua_2017/metadata.yml")$context, dataset_id, traits)
 #' }
-process_format_contexts <- function(my_list, dataset_id) {
+process_format_contexts <- function(my_list, dataset_id, traits) {
 
-  f <- function(x) {
+  f <- function(x, id) {
     tibble::tibble(
     context_property = x$context_property,
     category = x$category,
     var_in = x$var_in,
-    util_list_to_df2(x$values))
+    util_list_to_df2(x$values)
+    )  %>%
+    dplyr::mutate(dataset_id = dataset_id) %>%
+    dplyr::select(dplyr::any_of(
+        c("dataset_id", "context_property", "category", "var_in",
+          "find", "value", "description"))
+      )
   }
-
+  
   if (!is.na(my_list[1])) {
-    contexts <-
+    
+      contexts <-
       my_list %>%
-      purrr::map_df(f) %>%
-      dplyr::mutate(dataset_id = dataset_id) %>%
-      dplyr::select(dplyr::any_of(
-        c("dataset_id", "context_property", "category", "var_in",
-        "find", "value", "description"))
+      purrr::map(f, dataset_id)
+    
+    for (i in 1:length(contexts)) {
```

Comment:
I'm going to rework this, we may as well put all the stuff below into the function above, as it already works through each context. 

There's no need creating these groups with a map, then looping over them.